### ISSUE TITLE
Docs: Activation for nav pane and directly going to a page

### ISSUE THREAD

**Issue by crutkas** [Microsoft/WindowsTemplateStudio#482]
From #456, a user was interested in having his app go directly to a page, not using ShellPage.

We should provide guidance on
 - directly going to that page VS using shell page
 - warn of any issues this could cause
 - how to move back to shell page if the user clicks a button or something to go back to the main page

**Comment by crutkas** [comment]
feel like https://github.com/Microsoft/WindowsTemplateStudio/blob/master/docs/activation.md or a subpage for this should be where this lives

**Comment by touseefbsb** [comment]
well I havent really tried to add the functionality of going back to the mainpage yet, because specs of my app are not yet decided about that feature. But for now I can tell how to navigate to another page by skipping the **ShellPage** which is pretty simple and straightforward.

1. Comment out the **OnActivated** method within "App.xaml.cs"
2. override **OnFileActivated** method within "App.xaml.cs".
3. and within method **OnFileActivated** just navigate to the required page.

**Side Effects**
No activation service is created for the App because that is normally handled in **OnLaunched** method.

**Code**

` //protected override async void OnActivated(IActivatedEventArgs args)
        //{           
        //    await ActivationService.ActivateAsync(args);
        //}

        protected override void OnFileActivated(FileActivatedEventArgs args)
        {
            if (args.Files.Count > 0)
            {
                Type TargetType = typeof(CompactNowPlayingPage);

                var SingleFileToPlay = args.Files[0] as StorageFile;
                TargetType = typeof(CompactNowPlayingPage);

                Frame rootFrame = Window.Current.Content as Frame;
                if (rootFrame == null)
                {
                    // Create a Frame to act as the navigation context and navigate to the first page
                    rootFrame = new Frame();

                    // Place the frame in the current Window
                    Window.Current.Content = rootFrame;
                }
                
                rootFrame.Navigate(TargetType, SingleFileToPlay);

                Window.Current.Activate();
            }
        }`

To simply navigate back to the ShellPage, just try following on the target page.
`this.Frame.Navigate(typeof(ShellPage));` // without the activation service support though.

I will try including the activation service with it and will update you guyx if I am successful in achieving that.

**Comment by EricAtMSFT** [comment]
I'd love to see this addressed. The ShellPage seems weird to me. It's the default landing page yet it can't be accessed from the hamburger. I'd like the a bility to just have the app start on MainPage yet still inherit the nav stuff from ShellPage.

**Comment by mrlacey** [comment]
It feels as if it may have been better to add the Navigation Panel by overriding the styling of the default frame. This would have avoided the confusion of `ShellPage` being an actual `Page`. Such a change is too big to consider right now (for 1.2), but we do need a simple documented way to display a page without app wide chrome (which is basically what the nav menu is.)
This also applies to apps using tabs for top level nav, as may have reason to hide that on some occasions too.

**Comment by crutkas** [comment]
@EricAtMSFT great feedback.  To do your adjustment, i few that as a new issue outside the primary issue.  If this is something you feel strongly on, lets please open a new issue.

i agree with @mrlacey that this is a larger change for 1.2 and what the preferred pattern would be for Navigation View for FCU and future work with that control.

I think we should really think about this and what is the best way as this should also be under consideration for Xamarin support.

**Comment by mrlacey** [comment]
@EricAtMSFT I'd like to understand a bit more about your scenario. The Shellpage is really just a container for the Navigation/Hamburger menu. It doesn't have any content and so I don't understand what you'd want from being able to navigate to it from the menu. If this existed you'd just end up with nav menu inside nav menu, potentially indefinitely.

The ability to navigate to a new page without any of the app's standard navigation chrome is something I've needed in many apps in the past. We should have picked up this need earlier. This may be needed at startup (e.g. sign-in pages) or from within the app (e.g. open an embedded browser or video player "full screen").


Here's my thinking for this:
1. Add docs to show a workaround for a short term fix.
2. Create a [new issue](https://github.com/Microsoft/WindowsTemplateStudio/issues/697) for having the ability to navigate without the "shell" built into the generated code as a long term solution.

**Comment by EricAtMSFT** [comment]
@mrlacey For me, the ShellPage is n odd choice as the "MainPage" for this template, especially given that it's not even part of the hamburger.

I just want the app to start at one of the other pages I have created, that's included in the hamburger menu but I haven't been able to make that work.

TLDR. I'm open to either #1 or #2 mentioned above.

Thanks!

**Comment by mrlacey** [comment]
@EricAtMSFT If you just want to have the app open with a different page (other than the first in the list) simply change the View type reference in App.xaml.cs. So, in the below, `MainPage` could be changed to whatever you want.

    private ActivationService CreateActivationService()
    {
        return new ActivationService(this, typeof(Views.MainPage), new Views.ShellPage());
    }

If you want it without the menu visible then you'll need to wait for #697

**Comment by mvegaca** [comment]
I'm working on a PoC that use a most advanced NavigationService to reach more complex scenarios with a great usability.
The target on this is  PoC is to add these changes to the documentation to give to experienced developers to reach more complex scenarios.

The code can be checked out [here](https://github.com/mvegaca/WindowsTemplateStudio/tree/Issue482-navigation/poc/AdvancedNavigationPaneProject).

**Comment by touseefbsb** [comment]
@mvegaca  I think this is perfect way of doing it :)

**Comment by mvegaca** [comment]
PoC Updated to reach complex navigation scenarios. If someone has some comments on this PoC it could help me to improve the navigation system or reach scenarios in which I haven't thought.

[AdvancedNavigationPaneProject](https://github.com/mvegaca/WindowsTemplateStudio/tree/Issue482-navigation/poc/AdvancedNavigationPaneProject).

**Comment by touseefbsb** [comment]
I agre with @mvegaca  his work is definitely helpful as far as docs are concerned and I also agree with @mrlacey  that this should not replace NavigationService. as we dont wanna put extra complications to the app unless they are required. as these advanced scenarios are very specific and custom its best to have them in docs or maybe in the wizard as an advanced option? where the developer can check "advanced navigation" and they will be advised that it supports following 4,5 navigation scenarios and they should only use it if they know wht they r doing with it. otherwise by default the wizard will provide the simpler NavigationService.

**Comment by mvegaca** [comment]
Doc added on #2246

**Comment by sibille** [comment]
The doc shows how to modify the project to reach more advanced navigation scenarios. The code from the docs could be used to do an advanced navigation feature that could be used as basis for other templates.
